### PR TITLE
Relax httpx version to allow >= 0.18.0

### LINKED ISSUE #1463: httpx tests failing on httpx==0.23.1
Can we leave https://github.com/open-telemetry/opentelemetry-python-contrib/issues/1459 open until we actually fix the instrumentation to work with that version?

_Originally posted by @aabmass in https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1460#pullrequestreview-1186403709_

### PR DESCRIPTION
**Pull request by phillipuniverse**
# Description

The tests were failing in httpx > 0.23.1 because of this change in httpx - https://github.com/encode/httpx/pull/2407

Pretty easy workaround illustrated in [the changed tests in that PR](https://github.com/encode/httpx/pull/2407/files#diff-883169411234feeb94cdff4b5399a8a84b13fd72c41e7407f7bf9909b086ebe2R16-R17)

Fixes #1463, #1460 and #1459.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Automated tests caught the failure and validated the fix

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [x] Unit tests have been added
- [x] Documentation has been updated

### CODE REVIEW

**Comment by phillipuniverse** [comment]
This is the test failure:

```
=================================== FAILURES ===================================
__ TestStarletteManualInstrumentation.test_basic_post_request_metric_success ___

self = <tests.test_starlette_instrumentation.TestStarletteManualInstrumentation testMethod=test_basic_post_request_metric_success>

    def test_basic_post_request_metric_success(self):
        start = default_timer()
        expected_duration_attributes = {
            "http.flavor": "1.1",
            "http.host": "testserver",
            "http.method": "POST",
            "http.scheme": "http",
            "http.server_name": "testserver",
            "http.status_code": 405,
            "net.host.port": 80,
        }
        expected_requests_count_attributes = {
            "http.flavor": "1.1",
            "http.host": "testserver",
            "http.method": "POST",
            "http.scheme": "http",
            "http.server_name": "testserver",
        }
        self._client.post("/foobar")
        duration = max(round((default_timer() - start) * 1000), 0)
        metrics_list = self.memory_metrics_reader.get_metrics_data()
        for metric in (
            metrics_list.resource_metrics[0].scope_metrics[0].metrics
        ):
            for point in list(metric.data.data_points):
                if isinstance(point, HistogramDataPoint):
                    self.assertEqual(point.count, 1)
>                   self.assertAlmostEqual(duration, point.sum, delta=30)
E                   AssertionError: 34 != 3 within 30 delta (31 difference)

test_starlette_instrumentation.py:175: AssertionError
========================= 1 failed, 49 passed in 2.26s =========================
```

It's over in Starlette and maybe this is a flaky one?

**Comment by macieyng** [comment]
@lzchen @jeremydvoss please review this PR 🙏

File: instrumentation/opentelemetry-instrumentation-httpx/tests/test_httpx_integration.py [hunk 1]
```diff
@@ -68,7 +68,7 @@ async def _async_response_hook(
 ):
     span.set_attribute(
         HTTP_RESPONSE_BODY,
-        await response[2].aread(),
+        b"".join([part async for part in response[2]]),
```
Comment (jeremydvoss) [inline instrumentation/opentelemetry-instrumentation-httpx/tests/test_httpx_integration.py:71]: Could someone explain why in the part for part iteration is only needed for async when it is used for both in this pr?
https://github.com/encode/httpx/pull/2407/files

Comment (phillipuniverse) [reply instrumentation/opentelemetry-instrumentation-httpx/tests/test_httpx_integration.py:71]: Originally I had it applied the same in both just like the httpx PR. CI failed with a Pylint issue saying that the [part for part] expression is unnecessary. Pylint is right, no need to iterate through an iterator once again, .join() can accept response[2] as-is.

On the async one, we’re dealing with async code so we need to await each coroutine part to convert it into a list to pass to .join().

**Comment by macieyng** [comment]
@phillipuniverse can you take a look at the pipeline?

**Comment by phillipuniverse** [comment]
@ocelotl @macieyng done

**Comment by phillipuniverse** [comment]
The latest test failure looks like a flaky one in aiohttp

image

**Comment by phillipuniverse** [comment]
Unfortunately I can't manually re-run the build, @ocelotl can you?

**Comment by macieyng** [comment]
@phillipuniverse I'm trying to understand the consequences before this was marged. Assuming I had httpx==0.24.1 and opentelemetry-instrumentation-httpx==0.38b0 would it mean that my httpx was not instrumented due to that reason?

**Comment by phillipuniverse** [comment]
@macieyng yes that is probably the reason.

**Pull request merged**